x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/RestEsqlQueryAction.java

```diff
@@ -61,4 +63,30 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
     protected Set<String> responseParams() {
         return Set.of(URL_PARAM_DELIMITER, EsqlQueryResponse.DROP_NULL_COLUMNS_OPTION);
     }
+
+    static final String PRODUCT_ORIGIN = "x-elastic-product-origin";
+    static final String CLIENT_META = "x-elastic-client-meta";
+
+    static void defaultVersionForOldClients(EsqlQueryRequest esqlRequest, RestRequest restRequest) {
```

Comment:
nit/bikeshed: This could also be a method of `EsqlQueryRequest` since it just mutates its version.

Also, in the added tests we do not _really_ test the handling of RestEsqlQueryAction, the tests are based on EsqlQueryRequests. We could move the tests to EsqlQueryRequestsTests.

Reply:
Same as above - yeah, it totally could. But I, personally, think of it as a REST layer hack rather than a request hack.